libraries/botbuilder-dialogs-adaptive/src/jsonExtensions.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+
+import { DialogStateManager } from 'botbuilder-dialogs';
+import { ExpressionParser, ValueExpression } from 'adaptive-expressions';
+
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+export class JsonExtensions {
+    /**
+     * Replaces the binding paths in a JSON value with the evaluated results recursively. Returns the final JSON value.
+     * @param dc A scope for looking up variables.
+     * @param token A JSON value which may have some binding paths.
+     * @returns Deep data binding result.
+     */
+    public static replaceJsonRecursively(state: DialogStateManager, token: object): any {
+        if (typeof token === 'string') {
+            let text: string = token as string;
+            if (text.startsWith('{') && text.endsWith('}')) {
+                text = text.slice(1, text.length - 1);
+                const { value } = new ExpressionParser().parse(text).tryEvaluate(state);
+                return value;
+            }
```

Comment:
I don't think this block of code ever gets hit because string based expression bindings are wrapped with ${expression} not {expression}. The comment in the C# code is wrong.

Additionally the C# code doesn't do this step because it's the role of the underlying expression parser to parse these bindings.

Reply:
I have merged the master branch to update the binding logic.